tedconf/code-style-guides
# distribute rubocop as a gem

fixes #70. this PR packages our rubocop styles as a gem. 

## feedback requested

@tedconf/backenders I'd love some input on how this approach seems to everybody.

  1. do my installation instructions (below) work for you? if not, what went wrong?
  2. if you wanted to make a change to the rubocop rules, do you know how you'd go about that? (i can/will improve the docs on that if needed.)
  3. once we're packaging our rubocop rules as a gem, our projects won't all automatically get the latest styles (as we currently do since we're all sourcing rules from master over HTTP). projects will have to explicitly update the `ted_rubocop_rules` gem to get updated rules. does that sound ok?

## TODO

  - [x] our existing process for sourcing the rule set via HTTP (from raw.githubusercontent.com) should continue to work, but that needs a bit of verification. (i'm trying a symlink to avoid copy/pasting the file. not 100% sure that'll play nice with github.com.)
  - [x] better docs on how to do a new release
  - [ ] get a few projects to install and use a prerelease of this gem. (notes below.)

## how to test this out

1. add this to a project's Gemfile

```ruby
gem 'ted_rubocop_rules'
```

2. `bundle update --conservative rubocop` to make sure you have the version of rubocop required our gem.

3. update your `.rubocop.yml`

```yml
AllCops:
  TargetRubyVersion: 2.5 # adjust for your project

inherit_gem:
  ted_rubocop_rules: rubocop.yml

inherit_from: .rubocop_todo.yml # if you have one
```

4. regenerate your todo list. `bundle exec rubocop --auto-gen-config`
